src/utils/chartJs/CustomTooltip.jsx

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import { withStyles } from '@material-ui/core/styles';
+import Fade from '@material-ui/core/Grow';
+
+const styles = {
+  tooltip: {
+    background: 'rgba(0, 0, 0, .8)',
+    color: 'white',
+    borderRadius: '4px',
+    pointerEvents: 'none',
+  },
+  tooltipKey: {
+    display: 'inline-block',
+    width: '10px',
+    height: '10px',
+    marginRight: '10px',
+  },
+  table: {
+    borderCollapse: 'collapse',
+  },
+};
+
+function CustomTooltip({ classes, tooltipModel }) {
+  const titleLines = tooltipModel.title || [];
+  const bodyLines = tooltipModel.body
+    ? tooltipModel.body.map(bodyItem => bodyItem.lines)
+    : [];
+  const footerLines = tooltipModel.footer || [];
+
+  return (
+    <Fade in={tooltipModel.opacity === 1}>
+      <div
+        className={classes.tooltip}
+        style={{
+          padding: `${tooltipModel.yPadding}px ${tooltipModel.xPadding}px`,
+        }}
+      >
+        <table className={classes.table}>
+          <thead>
+            {titleLines.map(title => (
+              <tr>
+                <th>{title}</th>
+              </tr>
+            ))}
+          </thead>
+          <tbody>
```

Comment:
probably always empty, do not render

Reply:
@klahnakoski: Nope, it's never empty by default. I consoled log `tooltipModel`for you when a data point is hovered in [this gist](https://gist.github.com/2alin/2a1ef853d3dc9d251bb398b1f624c422). 

ChartJS (by default) uses `.title` for the x-values (in our case, the date), and builds `.body` using the dataset (series) label and the y-value (in our case, e.g. "Moto G5 (arm7): 257" ). So title and body are build by chartjs as usual, I'm only rendering them.

`.footer`was the one that was build by us (using something that chartjs calls 'fallback' in the docs) and because it contains anchors we are rendering them in a special way.

--------

Now we could build everything in one container (for example, everything in `.body`) but, we need to create empty callbacks for `.title`and `.footer` and create a long callback for `.body` that stores all the lines needed. And then we can render all the tooltip using only one map (bodyLines.map). What do you think? do you like this idea better?

Reply:
> ChartJS (by default) uses `.title` for the x-values (in our case, the date), and builds `.body` using the dataset (series) label and the y-value (in our case, e.g. "Moto G5 (arm7): 257" ). 

Ok.  Instead of rendering `body`, render the "dataset (series) label and the y-value" ourselves.  The idea is to avoid using the ChartJS entirely so no knowledge is required to understand the custom tooltip.  We may consider overriding the `body` function (`()=>null`) just so we do not run the default chartJS code 
 
> Now we could build everything in one container (for example, everything in `.body`) but, we need to create empty callbacks for `.title`and `.footer` and create a long callback for `.body` that stores all the lines needed. 

Yes, empty callbacks for `.title`and `.footer` are good. We will not need the `body` lines because we can get the info from the tooltipModel

    seriesName + " " +tooltipModel.dataPoints[0].yLabel

I have tried to find where `seriesName` can be pulled from (other than the `body`, but I can not seem to find it.

> And then we can render all the tooltip using only one map (bodyLines.map). What do you think? do you like this idea better?

I do not like this, but since I can not find where `seriesName` is in the chartJS datastructure, then I guess that's how it must be.

Reply:
> The idea is to avoid using the ChartJS entirely so no knowledge is required to understand the custom tooltip.

Got it!!



> but since I can not find where `seriesName` is in the chartJS datastructure

This!! was the false alarm I was going to talk with you last night (along as many other values needed related to it. But it's  a false alarm, because we can use

Reply:
> The idea is to avoid using the ChartJS entirely so no knowledge is required to understand the custom tooltip.

Got it!!



> but since I can not find where `seriesName` is in the chartJS datastructure

@klahnakoski This!! was the false alarm I was going to talk with you last night, as well as many other values needed related to it. But it's  a false alarm, because we can use  `series` from line 12: 
```const generateInitialOptions = series => {...} ```
it's passed all the way down to `tooltips.custom` :grin: